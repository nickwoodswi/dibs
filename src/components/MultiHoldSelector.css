.multi-date-selector{
    width: 100%;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    border-radius: 5px;
    border: 1px dotted red;
    padding: 15px;
    text-align: center;
}
.add-hold-range-button{
    width: 100%;
}
    .add-hold-range-button:active{
        color: white;
        background-color: red;
    }