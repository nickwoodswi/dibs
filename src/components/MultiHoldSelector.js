import React, { Component } from 'react'
import StartDateSelector from './StartDateSelector'
import EndDateSelector from './EndDateSelector' 
import HoldPreview from './HoldPreview'
import MultiHoldSelector from './MultiHoldSelector.css'

class MultiDateSelector extends Component {
  static defaultProps = {
    appState: {
      holds: []
    }
  }
    render() {
      return(
        <div className="multi-date-selector">
          <div className="range-selector">
            <StartDateSelector 
              appState={this.props.appState} 
              handleState={this.props.handleState} /><br/>
            <EndDateSelector 
              appState={this.props.appState} 
              handleState={this.props.handleState} /><br/>
          </div>
          <button className="add-hold-range-button"
              id="add-hold-range" 
              onClick={this.props.addHold}>
          +ADD HOLD RANGE</button><br/>
          <HoldPreview appState={this.props.appState} deleteHold={this.props.deleteHold} />
        </div>
      )
    }
  }
  
  export default MultiDateSelector