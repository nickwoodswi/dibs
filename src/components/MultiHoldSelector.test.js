import React from 'react';
import { render } from '@testing-library/react';
import MultiHoldSelector from './MultiHoldSelector';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders multihold selector component', () => {
    render(<MemoryRouter><MultiHoldSelector /></MemoryRouter>);
});

it('MultiHoldSelector renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><MultiHoldSelector /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});