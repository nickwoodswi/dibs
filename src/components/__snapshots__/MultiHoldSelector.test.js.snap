// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiHoldSelector renders the UI as expected 1`] = `
<div
  className="multi-date-selector"
>
  <div
    className="range-selector"
  >
    FROM:
    <br />
    <select
      className="month-selector"
      onChange={[Function]}
    >
      <option>
        January
      </option>
      <option>
        February
      </option>
      <option>
        March
      </option>
      <option>
        April
      </option>
      <option>
        May
      </option>
      <option>
        June
      </option>
      <option>
        July
      </option>
      <option>
        August
      </option>
      <option>
        September
      </option>
      <option>
        October
      </option>
      <option>
        November
      </option>
      <option>
        December
      </option>
    </select>
    <select
      className="day-selector"
      onChange={[Function]}
    />
    <select
      className="year-selector"
      onChange={[Function]}
    >
      <option>
        2020
      </option>
      <option>
        2021
      </option>
    </select>
    <br />
    THRU: 
    <br />
    <select
      className="end-month-selector"
      onChange={[Function]}
    >
      <option>
        January
      </option>
      <option>
        February
      </option>
      <option>
        March
      </option>
      <option>
        April
      </option>
      <option>
        May
      </option>
      <option>
        June
      </option>
      <option>
        July
      </option>
      <option>
        August
      </option>
      <option>
        September
      </option>
      <option>
        October
      </option>
      <option>
        November
      </option>
      <option>
        December
      </option>
    </select>
    <select
      className="end-day-selector"
      onChange={[Function]}
    />
    <select
      className="end-year-selector"
      onChange={[Function]}
    >
      <option>
        2020
      </option>
      <option>
        2021
      </option>
    </select>
    <div
      className="current-add"
    >
      <div
        className="current-add-header"
      >
        CURRENTLY ADDING HOLDS FROM:
      </div>
       
      , 
       thru 
       
      , 
    </div>
    <br />
    <br />
  </div>
  <button
    className="add-hold-range-button"
    id="add-hold-range"
  >
    +ADD HOLD RANGE
  </button>
  <br />
  <div
    className="multi-hold-preview"
  />
</div>
`;
